transport/src/main/java/io/netty/channel/DefaultChannelHandlerContext.java

```diff
@@ -744,7 +744,15 @@ private void invokeRead() {
         try {
             ((ChannelOutboundHandler) handler()).read(this);
         } catch (Throwable t) {
-            notifyHandlerException(t);
+            invokeExceptionCaughtFromOutbound(t);
+        }
+    }
+
+    private void invokeExceptionCaughtFromOutbound(Throwable t) {
+        if ((executionMask & MASK_EXCEPTION_CAUGHT) != 0) {
+            invokeExceptionCaught(t);
```

Comment:
Not `notifyHandlerException()`? It has the `inExceptionCaught()` checks, that seem like they apply here just like anywhere else.

Reply:
@ejona86 true... let me fix this